### PR TITLE
[6.x] Override (static) password key name

### PR DESCRIPTION
**Pull request by ahmedsayedabdelsalam**
There was a case when I needed to make the User model store the password in a different column rather than "password", unfortunately, this column can't be changed, so to make it more dynamic, I've added a method that can be overridden, where we can specify the name of the column that stores the password, and an attribute in the User model that has a default value for that column with a value = "password"

old: `if (Str::contains($key, 'password'))`

new: `if (Str::contains($key, $query->getModel()->getPasswordKeyName()))`

the new attribute: `protected $passwordKey = 'password';`

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -1330,6 +1337,29 @@ public function setKeyName($key)
...
+        return $this->passwordKey;
+    }
+
+    /**
+     * Set the password key for the model.
+     *
+     * @param  string  $key
+     * @return $this
+     */
+    public function setPasswordKeyName($key)
```
Comment (jmarcher) [inline src/Illuminate/Database/Eloquent/Model.php]: Do we really need this method? The table name will not change on runtime

**Review by jmarcher — CHANGES_REQUESTED** [review]
Also add some tests for the new code.

Edit: Fix the tests as well

**Comment by X-Coder264** [comment]
I don't see why we should add all these methods to the base Eloquent model class (which all Eloquent models extend) when this is very specific user model stuff? The `password` column is hardcoded and it would be nice to be more flexible about it, but this implementation is not the way to go IMO.

**Comment by svenluijten** [comment]
I agree with @X-Coder264, I think this code would fit better in [`Illuminate\Auth\Authenticatable`](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Auth/Authenticatable.php).

**Comment by ahmedsayedabdelsalam** [comment]
> I don't see why we should add all these methods to the base Eloquent model class (which all Eloquent models extend) when this is very specific user model stuff? The `password` column is hardcoded and it would be nice to be more flexible about it, but this implementation is not the way to go IMO.

@X-Coder264 , @svenluijten 

You're Right, thank you.
Which way Would be better : 
1- fix it in another commit 
2- undo these commits and add new one
3- close the PR and open a new one

**Comment by svenluijten** [comment]
You can add a commit to this PR @ahmedsayedabdelsalam.

File: src/Illuminate/Database/Eloquent/Model.php (hunk 1 above)
Comment (jmarcher) [reply src/Illuminate/Database/Eloquent/Model.php]: @ahmedsayedabdelsalam have any use cases for this?

**Comment by taylorotwell** [comment]
We wouldn't add anything about passwords to the base Eloquent model. What if my model is a street address?

**Comment by Abdalmoniem8** [comment]
.

**Comment by ahmedsayedabdelsalam** [comment]
I wrote a comment that I will fix the implementation as @X-Coder264 request
I didn't get why you closed the PR, Is using (password) statically in code more appropriate ?!

**Comment by Abdalmoniem8** [comment]
اضرب يا استاذ احمد

**Comment by svenluijten** [comment]
@ahmedsayedabdelsalam no worries, you can submit another PR with this same functionality except (as discussed) put in the `Authenticatable` class. I'm sure that'll be accepted no problem.

**Comment by ahmedsayedabdelsalam** [comment]
new PR [30161](https://github.com/laravel/framework/pull/30161)

**Pull request closed without merging**